src/main.ts

```diff
@@ -20,322 +20,66 @@
-      throw new Error(
-        `Options object ${JSON.stringify(options)} has invalid key "${key}"`
-      );
-    }
-  });
-}
-
-function _makeDefaultContext<T>(
-  cloudFunction: CloudFunction<T>,
-  options: ContextOptions,
-  triggerData?: T
-): EventContext {
-  let eventContextOptions = options as EventContextOptions;
-  const eventResource = cloudFunction.__trigger.eventTrigger?.resource;
-  const eventType = cloudFunction.__trigger.eventTrigger?.eventType;
-
-  const optionsParams = eventContextOptions.params ?? {};
-  let triggerParams = {};
-  if (eventResource && eventType && triggerData) {
-    if (eventType.startsWith('google.firebase.database.ref.')) {
-      let data: database.DataSnapshot;
-      if (eventType.endsWith('.write')) {
-        // Triggered with change
-        if (!(triggerData instanceof Change)) {
-          throw new Error('Must be triggered by database change');
-        }
-        data = triggerData.before;
-      } else {
-        data = triggerData as any;
-      }
-      triggerParams = _extractDatabaseParams(eventResource, data);
-    } else if (eventType.startsWith('google.firestore.document.')) {
-      let data: firestore.DocumentSnapshot;
-      if (eventType.endsWith('.write')) {
-        // Triggered with change
-        if (!(triggerData instanceof Change)) {
-          throw new Error('Must be triggered by firestore document change');
-        }
-        data = triggerData.before;
-      } else {
-        data = triggerData as any;
-      }
-      triggerParams = _extractFirestoreDocumentParams(eventResource, data);
-    }
-  }
-  const params = { ...triggerParams, ...optionsParams };
-
-  const defaultContext: EventContext = {
-    eventId: _makeEventId(),
-    resource: eventResource && {
-      service: cloudFunction.__trigger.eventTrigger?.service,
-      name: _makeResourceName(eventResource, params),
-    },
-    eventType,
-    timestamp: new Date().toISOString(),
-    params,
-  };
-  return defaultContext;
-}
-
-function _extractDatabaseParams(
-  triggerResource: string,
-  data: database.DataSnapshot
-): EventContext['params'] {
-  const path = data.ref.toString().replace(data.ref.root.toString(), '');
-  return _extractParams(triggerResource, path);
-}
-
-function _extractFirestoreDocumentParams(
-  triggerResource: string,
-  data: firestore.DocumentSnapshot
-): EventContext['params'] {
-  // Resource format: databases/(default)/documents/<path>
-  return _extractParams(
-    triggerResource.replace(/^databases\/[^\/]+\/documents\//, ''),
-    data.ref.path
-  );
+  return wrapV1<T>(cloudFunction as CloudFunctionV1<T>);
 }
 
 /**
- * Extracts the `{wildcard}` values from `dataPath`.
- * E.g. A wildcard path of `users/{userId}` with `users/FOO` would result in `{ userId: 'FOO' }`.
- * @internal
+ * The key differences between V1 and V2 CloudFunctions are:
+ * <ul>
+ *    <li> V1 CloudFunction's handler requires the 1st arg, but has an optional 2nd.
```

Comment:
I think the language here might be a bit off. When you say `V1 CloudFunction's handler` and `V1 CloudFunction's run handler` I get the impression that you're talking about the same thing. The V1CloudFunction _is_ a function, which behaves as you describe the "run handler". But there's the callback the customer provides is available at `<cloudfunction>.run` and that behaves as you call the "V1 CloudFunction's handler"